debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogFileCollector.java

```diff
@@ -264,9 +265,55 @@ private boolean isOpenThreadConsistent(RedoThread thread, Scn startScn, List<Log
         else {
             // Check whether the redo logs have the read position
             if (threadLogs.stream().noneMatch(log -> log.isScnInLogFileRange(startScn))) {
-                logException(String.format("Redo Thread %d is inconsistent; does not have a log that contains scn %s." +
-                        "A recent log switch may not have been archived by the Oracle ARC process yet.", threadId, startScn));
-                return false;
+                try {
+                    // Collect all archive logs for a given redo thread that has an SCN range that comes
+                    // after or includes the startScn plus the archive log that has a range that comes
+                    // immediately before the startScn
+                    final List<LogFile> allThreadArchiveLogs = getAllRedoThreadArchiveLogs(threadId);
+
+                    if (allThreadArchiveLogs.isEmpty()) {
+                        logException(String.format("Redo Thread %d is inconsistent; at least one archive log expected.", threadId));
+                        return false;
+                    }
+                    else if (allThreadArchiveLogs.stream().anyMatch(l -> l.isScnInLogFileRange(startScn))) {
+                        logException(String.format("Redo thread %d is inconsistent; does not have a log that conatins scn %s. " +
+                                "A recent log switch may not have been archived by the Oracle ARC process yet.", threadId, startScn));
+                        return false;
+                    }
+                    else {
+                        // Collects all archive logs for a given redo thread and returns the first archive log
+                        // with an SCN range that comes immediately before the log with a range that includes the startScn.
+                        final Optional<LogFile> logWithRangeBeforeStartScn = allThreadArchiveLogs.stream()
+                                .filter(l -> l.getFirstScn().compareTo(startScn) < 0 && !l.isScnInLogFileRange(startScn))
+                                .findFirst();
```

Comment:
@Naros What is the scenario when more than one file can match the filetr?

Reply:
There isn't one, would you rather I use `findAny()` instead @jpechane ?